Source/AwsCommonRuntimeKit/crt/CBOR.swift

```diff
@@ -0,0 +1,360 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+import AwsCCommon
+import Foundation
+
+/// CBOR Types. These types don't map one-to-one to the CBOR RFC.
+public enum CBORType: Equatable {
+    /// UINT64 type for positive numbers.
+    case uint64(_ value: UInt64)
+    /// INT64 type for negative numbers. If the number is positive, it will be encoded as UINT64 type.
+    case int(_ value: Int64)
+    /// Double type. It might be encoded as an integer if possible without loss of precision. Half-precision floats are not supported.
+    case double(_ value: Double)
+    /// Bytes type for binary data
+    case bytes(_ value: Data)
+    /// Text type for utf-8 encoded strings
+    case text(_ value: String)
+    /// Array type
+    case array(_ value: [CBORType])
+    /// Map type
+    case map(_ value: [String: CBORType])
+    /// Date type. It will be encoded as epoch-based date/time.
+    case date(_ value: Date)
+    /// Bool type
+    case bool(_ value: Bool)
+    /// Null type
+    case null
+    /// Undefined type
+    case undefined
+    /// Break type for indefinite-length arrays, maps, bytes, and text. For encoding, you should start the encoding
+    /// with `indef_*_start` and then end the encoding with this `indef_break` type. During decoding, you will get 
+    /// the `indef_*_start` type first, followed by N elements, and the break type at the end.
+    case indef_break
+    /// Indefinite Array Type
+    case indef_array_start
+    /// Indefinite Map Type
+    case indef_map_start
+    /// Indefinite Bytes Type
+    case indef_bytes_start
+    /// Indefinite Text Type
+    case indef_text_start
```

Comment:
trivial:
use the order that matches C/above
```suggestion
    /// Indefinite Bytes Type
    case indef_bytes_start
    /// Indefinite Text Type
    case indef_text_start
    /// Indefinite Array Type
    case indef_array_start
    /// Indefinite Map Type
    case indef_map_start
```